core/src/main/kotlin/xtdb/arrow/ListExpression.kt

```diff
@@ -0,0 +1,10 @@
+package xtdb.arrow
+
+import xtdb.vector.IVectorWriter
+import org.apache.arrow.vector.types.pojo.Field
+
+interface ListExpression {
+    fun field(): Field
```

Comment:
Not strictly necessary here, given that it's already returned by `:field-type`. Could probably drop it, unless there's any reason we might need to keep it/use it?